src/main/scala/io/spicelabs/goatrodeo/omnibor/strategies/Dotnet.scala

```diff
@@ -57,14 +59,26 @@ class DotnetState extends ProcessingState[SingleMarker, DotnetState] {
       item: Item,
       marker: SingleMarker
   ): (Vector[PackageURL], DotnetState) = {
-    if (fileStm == null) {
+    if (fileStmOpt.isEmpty || assemblyOpt.isEmpty) {
       Vector.empty -> this
     } else {
-      var purl = PackageURL("nuget", "", assembly.name.name, assembly.name.version.toString(), null, "")
+      val assembly = assemblyOpt.get
+      val purl = PackageURL("nuget", "", assembly.name.name, assembly.name.version.toString(), null, "")
       Vector(purl) -> this
     }
   }
 
+
+  private def streamForArtifact(artifact: ArtifactWrapper): FileInputStream = {
```

Comment:
There's a method `FileWalker.withinTempDir(...)` that does this and does the appropriate cleanup